dill/_shims.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+#
+# Author: Mike McKerns (mmckerns @caltech and @uqfoundation)
+# Copyright (c) 2008-2016 California Institute of Technology.
+# Copyright (c) 2016-2021 The Uncertainty Quantification Foundation.
+# License: 3-clause BSD.  The full license text is available at:
+#  - https://github.com/uqfoundation/dill/blob/master/LICENSE
+"""
+Provides shims for compatibility between versions of dill and Python.
+"""
+
+import inspect, sys
+
+_dill = sys.modules['dill._dill']
+
+# The values for the shims for this particular version of dill and Python.
+_dill_true_values = {}
+
+class Shim(object):
+    """
+    A wrapper object that refers to a shim object in the module if it exists and
+    to another object if it doesn't exist. This choice is made during the
+    unpickle step instead of the pickling process.
+    """
+    def __new__(cls, name, alternative, module=None):
+        if callable(alternative):
+            return object.__new__(_CallableShim)
+        else:
+            return object.__new__(cls)
+    def __init__(self, name, alternative, module=None):
+        if _dill_true_values is not None:
```

Comment:
There should be some documentation (maybe either at this module's header or footer) about what's the expected behavior for anything related to `_dill_true_values`.  I think it's potentially confusing enough that future developers will benefit from an explanation (and probably also include a reference to this PR).

Reply:
Doc LGTM